custom/sensors/CANSensor.java

```diff
@@ -11,89 +11,49 @@
 	 *        ID of CAN sensor (0x600 to 0x700, must correspond to a Teensy or similar)
-	 * @param modes
-	 *        Number of modes for the CAN sensor
-	 */
-	public CANSensor(String name, int id, int modes) {
-		super(name, id);
-		values = new int[modes];
-		ages = new long[modes];
-		for (int i = 0; i < modes; i++) {
-			values[i] = 0; // Should we make this a more obscure value (e.g. 2^32 - 1)?
-			ages[i] = System.currentTimeMillis();
-		}
-	}
-
-	/**
-	 *
-	 * @param name
-	 *        Name of CAN sensor
-	 * @param id
-	 *        ID of CAN sensor (0x600 to 0x700, should correspond to a Teensy or similar)
 	 */
 	public CANSensor(String name, int id) {
-		this(name, id, 1);
-	}
-
-	/**
-	 * Read an int from a CAN sensor with retries
-	 * Retries are now ignored
-	 *
-	 * @param mode
-	 *        Which mode to read the sensor in (interpreted by the Teensy)
-	 * @return
-	 * 		The integer the Teensy returned for that mode
-	 *
-	 * @throws InvalidSensorException
-	 *         If the available data is more than one second old,
-	 *         this function will throw an InvalidSensorException
-	 *         to indicate that.
-	 */
-	@Deprecated
-	public int read(int mode, int retryMax) throws InvalidSensorException {
-		return read(mode);
+		super(name, id);
+		values = new int[2];
+		values[0] = 0;
+		values[1] = 0;
+		age = System.currentTimeMillis();
 	}
 
 	/**
-	 * Read an int from a CAN sensor
+	 * Read the pair of ints from a CAN sensor
 	 *
-	 * @param mode
-	 *        Which mode to read the sensor in (interpreted by the Teensy)
 	 * @return
-	 * 		The integer the Teensy returned for that mode
+	 * 		The latest pair of integers from the sensor
 	 *
 	 * @throws InvalidSensorException
-	 *         If the available data is more than one second old,
+	 *         If the available data is more than one tenth of a second old,
 	 *         this function will throw an InvalidSensorException
 	 *         to indicate that.
 	 */
-	public int read(int mode) throws InvalidSensorException {
-		write(new byte[] {(byte) ((byte) mode & 0xFF), 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01}); // Write to trigger read
-		ByteBuffer rawData = readBuffer();
+	public int[] readSensor() throws InvalidSensorException {
+		ByteBuffer rawData = super.readBuffer();
 		if (rawData != null && rawData.remaining() > 7) {
 			rawData.rewind();
 			long data = Long.reverseBytes(rawData.getLong());
-			int value = (int) (data & 0xFFFFFFFF);
-			int msgMode = (int) (data >> 32);
-			if (msgMode == mode) {
-				ages[mode] = System.currentTimeMillis();
-				values[mode] = value;
-				return values[mode];
-			}
+			values[0] = (int) data & 0xFFFFFFFF;
+			values[1] = (int) (data >> 32) & 0xFFFFFFFF;
+			age = System.currentTimeMillis();
+			return values;
 		}
-		if (System.currentTimeMillis() - ages[mode] > MAX_AGE) {
-			throw new InvalidSensorException("CAN data oudated For CAN sensor " + getName() + " with ID " + messageID);
+		if (System.currentTimeMillis() - age > MAX_AGE) {
```

Comment:
The name `age` is misleading, especially since `MAX_AGE` is not the maximum value of `age`.